torchaudio/functional/functional.py

```diff
@@ -1444,3 +1446,82 @@ def resample(
                                               resampling_method, beta, waveform.device, waveform.dtype)
     resampled = _apply_sinc_resample_kernel(waveform, orig_freq, new_freq, gcd, kernel, width)
     return resampled
+
+
+def character_edit_distance(r: str, h: str) -> int:
+    """
+    Calculate the character level edit (Levenshtein) distance between two strings.
+
+    The function computes an edit distance allowing deletion, insertion and substitution.
+    The result is an integer. Users may want to normalize by the length of the reference.
+
+    This can be used to compute the edit distance for instance between two strings,
+    or two list of words. Note that, if a string and a list of words is provided, the distance
+    will be computed between the "string" sequence, and the "list of words" sequence.
+
+    Args:
+        r (str): the reference string to compare.
+        h (str): the hypothesis string (e.g. the predicted string) to compare.
+    Returns:
+        int: The distance between the reference and the hypothesis.
+    """
+
+    # Initialisation
+    dold = [i for i in range(len(h)+1)]
+    dnew = [0 for _ in range(len(h)+1)]
+
+    # Computation
+    for i in range(1, len(r) + 1):
+        dnew[0] = i
+        for j in range(1, len(h) + 1):
```

Comment:
`len(h) + 1` is called three times. Please put it in a variable.

Reply:
It is fixed [here](https://github.com/pytorch/audio/pull/1601/files#diff-f69839488ce70c39d8c3db8c87790ac963364faf8aefcb38ff5b026ee9dca8c7R1498)